src/main/host/syscall/format.rs

```diff
@@ -193,16 +193,28 @@ impl SyscallPtrDisplay for SyscallPtr<*const i8> {
         // much at once (for example bytes past the null byte), we may try to read from an invalid
```

Comment:
Orthogonal to this PR, but [memory_ref_prefix](https://shadow.github.io/docs/rust/shadow_rs/host/memory_manager/struct.MemoryManager.html#method.memory_ref_prefix) or [copy_prefix_from_ptr](https://shadow.github.io/docs/rust/shadow_rs/host/memory_manager/struct.MemoryManager.html#method.copy_prefix_from_ptr) might be useful here.

Reply:
> will the entire function return an error, or will it return something with access to bytes 0-30?

It's meant to do the latter.

`readv_ptrs` calls `readv_iovecs`, which calls `process_vm_readv`. `process_vm_readv` supports partial reads.

Reply:
Okay I think I see, so `memory_ref_prefix` will return an error if it could not copy any memory, otherwise it will return the memory read up until a page boundary where the copy read fails.

Reply:
I'll make a new PR for this.